### ISSUE TITLE
Cloudinary media library: can't log in

### ISSUE THREAD

**Issue by kennedyrose** [netlify/netlify-cms#1999]
Whenever setting the media library to cloudinary I'm met with this when trying to upload:

![screen shot 2019-01-10 at 3 59 12 pm](https://user-images.githubusercontent.com/1924910/50999780-c74e1300-14f0-11e9-97c5-dc00f98ea02c.png)

I've checked my cloud name and API key and they're both correct. It seems like unsigned uploads might need to be enabled in Cloudinary, but I can't do this without being able to set an "upload preset" somewhere in the Cloudinary widget. Am I doing something wrong? Is there a way to get uploads to work without signing into Cloudinary?

**Comment by erquhart** [comment]
The Cloudinary media library doesn't support unsigned uploads, the user would have to be signed in to Cloudinary.

**Comment by danhalliday** [comment]
Oh, you have to be signed in to these image services to use them? That makes these integrations pretty useless for my use case (client work), as I won't have much luck with many of them explaining they have to be logged into two services.

I don't recall reading that in the docs, perhaps it could be made clearer?

**Comment by talves** [comment]
@danhalliday A PR is always welcome to improve the documentation to make it clearer. Please let us know if you need help with where to find these documents in the project.

**Comment by kennedyrose** [comment]
It looks like there's a way to pass an [authentication signature](https://cloudinary.com/documentation/media_library_widget#2_optional_generate_the_authentication_signature) to the widget to automatically login. Would it be possible to set this up with Netlify Identity somehow?

**Comment by erquhart** [comment]
You could probably write a Netlify Function that checks Identity to ensure the user is logged in and has proper access, and then generates and sends back the auth signature. You would need to set the Cloudinary instance secret as an environment variable on the Netlify site.

Ideally the CMS would accept an API endpoint (URL to the Netlify Function) in the config, but being dependent on Netlify Identity requires collaboration between backends and media libraries, and that's something we need to think through. I'm not even certain all of this complexity should be coordinated by Netlify CMS. We should discuss this further.

I _think_ there's a way for you to do this today, but it would require run the Netlify Identity Widget manually. It would look something like this (again, haven't actually tried it):

1. Create the Netlify Function described above
2. Add the API Secret as an env variable to the Netlify site
3. In your code, set `window.CMS_MANUAL_INIT` to true before importing the CMS module, or before the CMS script tag if you're using it that way. This will stop the CMS from auto initializing: https://www.netlifycms.org/docs/beta-features/#manual-initialization
4. Add a [`login` event](https://github.com/netlify/netlify-identity-widget#usage) handler to the Netlify Identity Widget, which calls the Function, passing in the cloud name and api key and storing the returned signature
5. [Manually authenticate](https://github.com/netlify/netlify-identity-widget#usage) using the Identity Widget
6. Finally, [initialize the CMS](https://www.netlifycms.org/docs/beta-features/#manual-initialization), passing in a partial CMS configuration object containing the obtained signature, which the [Cloudinary docs] require to be set on the `signature` property. The CMS configuration object that you pass in will have your `config.yml` configuration merged over it, so the object you pass in during manual initialization will only need to contain the signature. Manual init will look something like this:

```js
import CMS, { init } from 'netlify-cms'

init({
  config: {
    media_library: {
      config: {
        signature: <signature>,
      },
    },
  },
})
```

Hope that makes even a little bit of sense. If you try this, don't hesitate to ask questions here or in [Gitter](https://gitter.im/netlify/netlifycms), the community is quite active and helpful.

**Comment by zslabs** [comment]
Has there been any improvements to this workflow? I absolutely love the integration feature by itself, but it seems quite convoluted to have that much setup involved ;or at the very least, not documented well on the CMS's official opinion on integration for teams.

**Comment by erquhart** [comment]
Not yet, still need to be logged in. Definitely want to see this improved, agreed.

**Comment by brianzelip** [comment]
Are other people actually able to access their Cloudinary account via Netlify CMS? I'm not able to login, just keep getting stuck in an endless loop of showing the same screenshot.

Not sure why this is happening, I have confirmed that all Cloudinary credentials are correct, and am currently logged into Cloudinary in another browser tab.

![Imgur](https://i.imgur.com/Iw3btnU.png)

**Comment by erezrokah** [comment]
Hi @brianzelip, are there any error/warnings in the browser console log?

**Comment by brianzelip** [comment]
Thanks for reaching out @erezrokah.

Since posting earlier, I logged in to both cloudinary and my netlify cms from a new computer, and the cloudinary authentication worked as expected. Not sure why it wasn't working from my normal dev computer - maybe a cache issue or something? If I continue to have problems on my dev computer I'll reach out in the Slack channel.

**Comment by zslabs** [comment]
A coworker also had this happen for them just in Chrome - Firefox worked fine. We never did figure out what was going on there; no console errors.

**Comment by erezrokah** [comment]
No sure, but [this](https://blog.chromium.org/2019/10/developers-get-ready-for-new.html) might be related. Will have to look further on how the Cloudinary lib is working.

**Comment by mischa-s** [comment]
just hit this today - wasn't working on firefox (normal or private mode) and then worked on chrome, seemingly the opposite of @zslabs experience, but I guess the tl;dr is it appears changing browser routes around whatever is happening

**Comment by brianzelip** [comment]
I've been working with it more, and I think it has to do with being logged in to the same cloudinary account on two diff browsers or machines.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by casperleerink** [comment]
Same issue here

**Comment by erezrokah** [comment]
Tried reproducing in 3 different browsers (Chrome, Brave and Firefox) and managed to login with all of them in the same time.
If there are no console errors, can someone share the browser network traffic during login (with `Preserve log` option checked?
Make sure to go to `https://cloudinary.com/users/logout` first.

Also found this thread https://support.cloudinary.com/hc/en-us/community/posts/360006487280-Media-Library-Widget-has-stopped-working

**Comment by erezrokah** [comment]
Closing, please re-open/comment if problem persists possibly with the additional information I requested.

**Comment by magneticscrolls** [comment]
This problem only happens on Safari consistently. Have tried clearing all caches and restarting. 
Still exists, very frustrating.

**Comment by erezrokah** [comment]
@magneticscrolls, can you try disabling "prevent cross site tracking" in Safari as suggested here:
https://support.cloudinary.com/hc/en-us/community/posts/360006487280/comments/360000889979

**Comment by magneticscrolls** [comment]
@erezrokah Excellent thanks, that worked!!
Although it only works if you log into Cloudinary first. 
Logging into Netlify CMS Admin, then clicking Sign-In on the empty dialog never works.

Could we just add a note to the Readme file at:
https://github.com/netlify/netlify-cms/tree/master/packages/netlify-cms-media-library-cloudinary

Just that this issue has been around for 8 months and must affect lots of end-users would use Safari.

**Comment by erezrokah** [comment]
@magneticscrolls just saying there is a "Edit this page" link here https://www.netlifycms.org/docs/cloudinary/ 😉

**Comment by brianzelip** [comment]
Just dropping a note here for folks to check their browser privacy extensions! I just ran into this issue again, checked this thread for any more tips, then checked my Privacy Badger extension. The extension was blocking 3 cloudinary apis. DOH!

**Comment by Jarmos-san** [comment]
So, is there a fix for it yet? I'm on Microsoft Edge & can't log in to Cloudinary through the CMS since the redirect never really works out. Would love to share more information on it, but I just need some guidance here since frontend stuff isn't my forte.

**Comment by Eric-Alain** [comment]
Not sure if this is helpful at all, but I had a similar issue to this, though not exactly as described above.

For me, it would present me with an option to sign in, but whenever I would click, nothing would happen, the screen would blink and I'd be stuck on the cloudinary sign in page. 

Anyways, I was using the brave browser with shields up, turning them off for the Netlify CMS page (and presumably any other kind of ad/pop up blocker) might help. 

Seems stupid, but I figured I'd mention it in case anyone else has overlooked the obvious, like I did.
